### PR TITLE
feat: icons for chat identification

### PR DESCRIPTION
**Pull request by berry-13**
## added an icon to the left of each chat to better identify the endpoint (they are in order from bottom to top)

image

### Before:

image

### After:

image

### Before:

image

### After:

image

## Change Type

- [x] New feature (non-breaking change which adds functionality)

## Testing

check if all the icon render and works correctly

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] I have made pertinent documentation changes
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes
- [x] Any changes dependent on mine have been merged and published in downstream modules.

### CODE REVIEW

**Comment by danny-avila** [comment]
Maybe I'm being nitpicky, but it looks like the title text is now further to the right from the icons with your changes. any way we can have more space for the title? 

image

**Comment by Berry-13** [comment]
Done!

image

File: client/src/components/Endpoints/getMinimalIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  PluginMinimalistIcon,
+  BingAIMinimalistIcon,
+  PaLMinimalistIcon,
+  AnthropicMinimalistIcon,
+} from '~/components/svg';
+import { useAuthContext } from '~/hooks';
+import { cn } from '~/utils';
+
+// Define the Props interface
+interface IconProps {
```
Comment (danny-avila) [inline client/src/components/Endpoints/getMinimalIcon.tsx]: repeated here and in getIcon, should be imported from ~/common

File: client/src/components/Endpoints/getMinimalIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          alt="avatar"
+        />
+      </div>
+    );
+  } else if (!isCreatedByUser) {
+    const { endpoint, error } = props;
+
+    let icon, name;
+    if (endpoint === 'azureOpenAI') {
+      const { chatGptLabel } = props;
```
Comment (danny-avila) [inline client/src/components/Endpoints/getMinimalIcon.tsx]: repetitive code deconstructing the same variables. I would prefer you not use the chained if/else-if statements in favor of a map. see client/src/components/Input/ModelSelect/ModelSelect.tsx for an example

File: client/src/components/Endpoints/getIcon.tsx [hunk 3]
```diff
@@ -38,17 +53,17 @@ const getIcon = (props) => {
       icon = <GPTIcon size={size * 0.7} />;
       bg = 'linear-gradient(0.375turn, #61bde2, #4389d0)';
       name = chatGptLabel || 'ChatGPT';
-    } else if (endpoint === 'openAI' || (endpoint === 'gptPlugins' && message)) {
+    } else if (endpoint === 'openAI') {
```
Comment (danny-avila) [inline client/src/components/Endpoints/getIcon.tsx]: do you mind fixing getIcon to match getMinimalIcon in structure? we can take this opportunity to clean up the code here

**Review by danny-avila — CHANGES_REQUESTED** [review]

File: client/src/components/Endpoints/getIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+          }
+          alt="avatar"
+        />
+      </div>
+    );
+  } else {
+    const { endpoint, error } = props;
+    let icon, bg, name;
+
+    switch (endpoint) {
```
Comment (danny-avila) [inline client/src/components/Endpoints/getIcon.tsx]: switch/case should never be used, you won't find it in this project: https://javascript.plainenglish.io/why-you-should-avoid-switch-statements-in-javascript-ce07d5b60da4

File: client/src/components/Endpoints/getIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+      case 'azureOpenAI': {
+        const { chatGptLabel } = props;
+        icon = <GPTIcon size={size * 0.7} />;
+        bg = 'linear-gradient(0.375turn, #61bde2, #4389d0)';
+        name = chatGptLabel || 'ChatGPT';
+        break;
+      }
+
+      case 'openAI': {
+        const { chatGptLabel } = props;
```
Comment (danny-avila) [inline client/src/components/Endpoints/getIcon.tsx]: also you are still running in the problem I outlined.

```js
const { chatGptLabel } = props;
...
const { chatGptLabel } = props;
...
const { chatGptLabel } = props;
...
const { chatGptLabel } = props;
```

I know this was here before but now is the opportunity to fix it

File: client/src/components/Endpoints/getMinimalIcon.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+          }
+          alt="avatar"
+        />
+      </div>
+    );
+  } else {
+    const { endpoint, error } = props;
+    let icon, name;
+
+    switch (endpoint) {
```
Comment (danny-avila) [inline client/src/components/Endpoints/getMinimalIcon.tsx]: no switch/case, use a map like in the original example I gave

**Review by danny-avila — CHANGES_REQUESTED** [review]

**Comment by Berry-13** [comment]
in the meantime I also removed some unnecessary functions in getMinimalistIcon

File: client/src/components/Endpoints/getIcon.tsx [hunk 7]
```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { Plugin, GPTIcon, AnthropicIcon } from '~/components/svg';
+import { useAuthContext } from '~/hooks';
+import { cn } from '~/utils';
+import { IconProps } from '~/common';
+
+const getIcon: React.FC<IconProps> = (props) => {
+  const { size = 30, isCreatedByUser, button, model = true, endpoint, error, jailbreak } = props;
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
```
Comment (danny-avila) [inline client/src/components/Endpoints/getIcon.tsx:10]: One last thing, I just noticed this eslint disable line. This and the minimalist function are already structured like React functional components so we should just make them. Rename it to Icon.tsx and the function as well, remove the comment and treat it like a regular React.FC(which it even has a type for).

**Review by danny-avila — APPROVED** [review]

File: client/src/components/Endpoints/getIcon.tsx (hunk 7 above)
Comment (danny-avila) [inline client/src/components/Endpoints/getIcon.tsx]: One last thing, I just noticed this eslint disable line. This and the minimalist function are already structured like React functional components so we should just make them. Rename it to Icon.tsx and the function as well, remove the eslint disable line, and treat it like a regular React.FC(which it even has a type for).

**Review by danny-avila — CHANGES_REQUESTED** [review]

**Comment by danny-avila** [comment]
thank you, the code is a lot cleaner now. I think the change to Icon.tsx and MinimalistIcon.tsx as React FC's is all that's left. Good work on this!

**Review by danny-avila — APPROVED** [review]

**Pull request merged**